### PR TITLE
 Storybook example for color selector with map layer

### PR DESCRIPTION
**Pull request by shruthirai**

### CODE REVIEW

**Review by nilscb — APPROVED** [review]

File: react/src/lib/components/DeckGLMap/layers/map/mapLayer.stories.tsx [hunk 1]
```diff
@@ -935,3 +940,98 @@ ColorMapRange.parameters = {
...
+    return (
+        <DeckGLMap {...args} layers={updatedLayerData}>
+            {
+                // @ts-expect-error This is demonstrated to work with js, but with ts it gives error
+                <View id="view_1_2D">
+                    <div style={{ marginTop: 50 }}>
+                        <ColorLegend
+                            min={min}
+                            max={max}
+                            colorNameFromSelector={colorNameFromSelector}
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/map/mapLayer.stories.tsx:1004]: Looks very nice!

Suggested improvement: would it be possible to generate the current colorMapFunction from the ColorLegend component?

**Review by hkfb — APPROVED** [review]

**Review by hkfb — COMMENTED** [review]
If I choose a discrete color scale, then the color legend disappears, and I get this error in the console:

> ```
> DiscreteLegend.js:199 TypeError: Cannot convert undefined or null to object
>     at Function.entries (<anonymous>)
>     at DiscreteLegend.js:89:1
>     at step (DiscreteLegend.js:43:1)
>     at Object.next (DiscreteLegend.js:24:1)
>     at DiscreteLegend.js:18:1
>     at new Promise (<anonymous>)
>     at push../node_modules/@emerson-eps/color-tables/dist/component/Legend/DiscreteLegend.js.__awaiter (DiscreteLegend.js:14:1)
>     at discreteLegend (DiscreteLegend.js:65:1)
>     at DiscreteLegend.js:62:1
>     at invokePassiveEffectCreate (react-dom.development.js:23487:1)
> ```

**Comment by shruthirai** [comment]
> If I choose a discrete color scale, then the color legend disappears, and I get this error in the console:
> 
> > ```
> > DiscreteLegend.js:199 TypeError: Cannot convert undefined or null to object
> >     at Function.entries (<anonymous>)
> >     at DiscreteLegend.js:89:1
> >     at step (DiscreteLegend.js:43:1)
> >     at Object.next (DiscreteLegend.js:24:1)
> >     at DiscreteLegend.js:18:1
> >     at new Promise (<anonymous>)
> >     at push../node_modules/@emerson-eps/color-tables/dist/component/Legend/DiscreteLegend.js.__awaiter (DiscreteLegend.js:14:1)
> >     at discreteLegend (DiscreteLegend.js:65:1)
> >     at DiscreteLegend.js:62:1
> >     at invokePassiveEffectCreate (react-dom.development.js:23487:1)
> > ```

Yes, have seen this error but forgot to fix it, will fix it.

**Comment by shruthirai** [comment]
Also fixes the bugs :

1) https://github.com/equinor/webviz-subsurface-components/issues/1353
2) https://github.com/equinor/webviz-subsurface-components/issues/1340 (Confirmed using this example : https://equinor.github.io/webviz-subsurface-components/storybook-static/?path=/story/deckglmap-components-colorlegends-discretecolorlegend--discrete-template)

**Comment by hkfb** [comment]
LGTM :+1:

**Pull request merged**